From 0e843062ac3916ee0a36f59ba53caabba64cb062 Mon Sep 17 00:00:00 2001
From: Matus Valo <matusvalo@gmail.com>
Date: Wed, 15 Mar 2023 21:02:10 +0100
Subject: [PATCH] Declare the function as noexcept when it is not raising
 exception.

Cython 3 will require noexcept keyword in the declaraction of the function not raising
an exceptions.
---
 pyproject.toml      | 2 +-
 src/krb5/_creds.pyx | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/krb5/_creds.pyx b/src/krb5/_creds.pyx
index a4a8a33..5b550e6 100644
--- a/src/krb5/_creds.pyx
+++ b/src/krb5/_creds.pyx
@@ -263,7 +263,7 @@ cdef krb5_error_code prompt_callback(
     const char *banner,
     int num_prompts,
     krb5_prompt *prompts,
-) with gil:
+) noexcept with gil:
     try:
         prompter = <Krb5Prompt>data
 
